modules/collections/client/controllers/collections.controllers.js

```diff
@@ -1,139 +1,154 @@
-            self.actionCancel = 'Cancel';
-            self.ok = function() {
-              $uibModalInstance.close($scope.project);
-            };
-            self.cancel = function() {
-              $uibModalInstance.dismiss('cancel');
-            };
-          },
-          controllerAs: 'self',
-          scope: $scope,
-          size: 'md',
-          windowClass: 'modal-alert',
-          backdropClass: 'modal-alert-backdrop'
-        });
-        modalDocView.result.then(function(/* res */) {
-          moveFunc();
+      // Shorthand for transitioning to Edit. This is interchangeable with $state.reload() *if* we're already in Edit (not in Create).
+      $scope.goToEdit = function() {
+        $state.transitionTo('p.collection.edit', { projectid: project.code, collectionId: collection._id }, {
+          reload: true, inherit: false, notify: true
         });
-      };
+      }
 
-      var goToList = function() {
+      // Shorthand for transitioning to List.
+      $scope.goToList = function() {
         $state.transitionTo('p.collection.list', { projectid: project.code }, {
           reload: true, inherit: false, notify: true
         });
-      };
+      }
 
-      var reloadEdit = function() {
-      // want to reload this screen, do not catch unsaved changes (we are probably in the middle of saving).
-        $scope.allowTransition = true;
-        $state.reload();
+      // Called after using the Document Manager modal.
+      $scope.updateDocuments = function(updatedDocuments, docType) {
+        var alternate = docType == 'main' ? 'other' : 'main';
+        var duplicates = $scope.findDuplicates(updatedDocuments, $scope.modifiedDocuments[alternate]);
+        if (duplicates.length) {
+          // If duplicates are found...
+          var errMsg = '';
+          _.forEach(duplicates, function(doc) {
+            errMsg += ('<br/> - ' + doc.displayName);
+            updatedDocuments = _.without(updatedDocuments, doc);
+            // Remove them from the updated documents list, and alert the user.
+          })
+          AlertService.error('The following document(s) already exist in "' + (docType == 'main' ? 'Related' : 'Main') + ' Documents" and have not been added:' + errMsg, 7000, false);
+        }
+        $scope.modifiedDocuments[docType] = _.map(updatedDocuments)
+        $scope.tableParams[docType] = new NgTableParams({ sorting: { sortOrder: 'asc' }, count: docType == 'main' ? 5 : 10 }, { dataset: $scope.modifiedDocuments[docType] });
+        $scope.validationFlags.docsChanged = true;
+        if ($scope.validationFlags.documentsTabInvalid) {
+          $scope.checkDocumentsField();
+        }
       };
 
-      $scope.otherDocsReordered = function() {
-        reloadEdit();
-      };
+      // Helper function to check if user has selected a document for both 'main' and 'other'.
+      $scope.findDuplicates = function(docs1, docs2) {
+        var duplicateFiles = [];
+        _.forEach(docs1, function(doc1) {
```

Comment:
Nit.: what about using [_.union()](https://lodash.com/docs#union) for this purpose?

Reply:
_.union() wouldn't work, because I want an intersection. I've just learned, however, that lodash has a _.intersection() function. I'll try implementing it, but it uses Same Value Zero for comparison, which may or may not work with this.